src/Microsoft.OpenApi/Services/OpenApiComparerBase.cs

```diff
@@ -138,6 +166,125 @@ internal void Compare<TEnum>(Enum source, Enum target, ComparisonContext compari
+
+            if (sourceReference.Id != targetReference.Id || sourceReference.Type != targetReference.Type)
+            {
+                WalkAndAddOpenApiDifference(
+                    comparisonContext,
+                    OpenApiConstants.DollarRef,
+                    new OpenApiDifference
+                    {
+                        OpenApiDifferenceOperation = OpenApiDifferenceOperation.Update,
+                        SourceValue = sourceReference,
+                        TargetValue = targetReference,
+                        OpenApiComparedElementType = typeof(OpenApiReference)
+                    });
+
+                return;
+            }
+
+            var source = (TReference) comparisonContext.SourceDocument.ResolveReference(
+                sourceReference);
+
+            var target = (TReference) comparisonContext.TargetDocument.ResolveReference(
+                targetReference);
+
+            comparisonContext
+                .GetComparer<TReference>()
+                .Compare(source, target, comparisonContext);
+        }
+
+        /// <summary>
+        /// Compares <see cref="IDictionary{TKey,TValue}"/> where TKey is <see cref="string"/> and TValue is
+        /// <see cref="string"/>.
+        /// </summary>
+        /// <param name="source">The source.</param>
+        /// <param name="target">The target.</param>
+        /// <param name="comparisonContext">The context under which to compare the objects.</param>
+        internal void Compare(IDictionary<string, string> source, IDictionary<string, string> target,
+            ComparisonContext comparisonContext)
+        {
+            if (source == null && target == null)
+            {
+                return;
+            }
+
+            if (source == null || target == null)
+            {
+                comparisonContext.AddOpenApiDifference(
+                    new OpenApiDifference
+                    {
+                        OpenApiDifferenceOperation = OpenApiDifferenceOperation.Update,
+                        SourceValue = source,
+                        TargetValue = target,
+                        OpenApiComparedElementType = typeof(IDictionary<string, T>),
+                        Pointer = comparisonContext.PathString
+                    });
+
+                return;
+            }
+
+            var newKeysInTarget = target.Keys.Except(source.Keys).ToList();
+
+            foreach (var newKeyInTarget in newKeysInTarget)
+            {
+                comparisonContext.AddOpenApiDifference(
+                    new OpenApiDifference
+                    {
+                        OpenApiDifferenceOperation = OpenApiDifferenceOperation.Add,
+                        TargetValue = new KeyValuePair<string, string>(
+                            newKeyInTarget,
+                            target[newKeyInTarget]),
+                        OpenApiComparedElementType = typeof(KeyValuePair<string, string>)
+                    });
```

Comment:
Based on this [http://jsonpatch.com/](http://jsonpatch.com/), this should be

                WalkAndAddOpenApiDifference(
                    comparisonContext,
                    newKeyInTarget,
                    new OpenApiDifference
                    {
                        OpenApiDifferenceOperation = OpenApiDifferenceOperation.Add,
                        TargetValue = target[newKeyInTarget],
                        OpenApiComparedElementType = typeof(string)
                    });

Reply:
I do not see a dictionary example for json patch , so i am not sure if this is correct, can you point me to a dictionary example that you might have found?

---
In reply to: [223511856](https://github.com/Microsoft/OpenAPI.NET/pull/328#discussion_r223511856) [](ancestors = 223511856)

Reply:
C# Dictionary should be treated as a JSON object.

The first example on [http://jsonpatch.com/](http://jsonpatch.com/)
See the second line:

 { "op": "add", "path": "/hello", "value": ["world"] }

and the third line:

{ "op": "remove", "path": "/foo" }

---
In reply to: [223857178](https://github.com/Microsoft/OpenAPI.NET/pull/328#discussion_r223857178) [](ancestors = 223857178,223511856)